server/src/chatroomUsers.ts

```diff
@@ -0,0 +1,118 @@
+import express, { query } from "express";
+import { IChatRoom, IPinnedItem, error } from "./types";
+import db from "./firebase";
+import { IBackgroundState } from "./types";
+
+const collection = db.collection("chatroomUsers");
+
+const chatroomUserRouter = express.Router();
+
+//get all chatroom users
+chatroomUserRouter.get("/get-all", async (req, res) => {
+    const doc = await collection.get();
+
+    res.status(200).send(doc.docs.map(doc => doc.data()));
+})
+
+//create new chatroom user
+chatroomUserRouter.post("/add-user", async (req, res) => {
```

Comment:
not a great naming convention, POST already implies creating a new record or some action not directly affecting the resource.
In this case it could be chatroomUser.post('user')